### ISSUE TITLE
Emulator; Cannot enable CPU VT

### ISSUE THREAD

**Issue by SolAZDev** [kamildzi/Tizen4Docker#15]
Good Evening once again. I have noticed that I cannot enable the CPU VT option in the emulator configuration. It responds;

```"Your system cannot support HW Virtualization. Try installing KVM (ubuntu) or HAX (windows and mac)"```

I've tried installing it but it didn't seem to work, and the debian image seems to reset every time it's open as it's never persistent (or at least when I tried it). I tried adding `qemu-kvm libvirt-clients libvirt-daemon-system` to the emulator dependencies in the Dockerfile but it didn't seem to work, as while the packages are installed, the option is still 'invalid'

PS: Running apt update also returned this;
```W: GPG error: http://dl.google.com/linux/chrome/deb stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 78BD65473CB3BD13
E: The repository 'http://dl.google.com/linux/chrome/deb stable InRelease' is not signed.
```

**Comment by kamildzi** [comment]
Hello. 

In general, this error states that your system lacks the support of KVM virtualization. 
Tizen4Docker container runs in privileged mode, therefore if your host system supports KVM I believe it should be supported also inside docker container. 

Please note, that support for KVM feature needs to be provided by both: the hardware (CPU) and the system. 
Please check following article on Arch Linux wiki (it should be fine for most of the distros): 
https://wiki.archlinux.org/index.php/KVM
Take a special note on "Checking support for KVM" paragraph. 

If the above article doesn't help you, could you please post the output of following commands?
`LC_ALL=C lscpu`
`uname -a`
`lsmod | grep kvm`

I tried to reproduce this error on my side. Unfortunately the only way I could achieve this, was putting docker container into non-privileged mode (changing `docker-compose.yml` setting to `false`). 

PS. You don't need to worry about the unsigned Google Chrome repository. This is fine. The cause of this error is the way the docker image is built. It may hurt ones eyes a little, but it doesn't break anything. :)

**Comment by kamildzi** [comment]
Hello.

I think that I have reproduced and solved the issue on my side. 
I needed to use a different machine to get this error. 
It happened to be a permission issue - docker user lacked permissions for '/dev/kvm' device. 

Could you please try the latest version from the master branch? 
Please note that new setting (`KVM_GID`) was added to `.env.example` file. 
You will need to merge it to your current settings at `.env`.

**Comment by SolAZDev** [comment]
I apologize I took longer than I anticipated, but the new changes worked marvelously, CPU VT was automatically enabled. Although I did get the message `Extra package installation is failed. sap must be installed MANUALLY!` I was able to compile and test out a sample project. So I think that solves this particular issue.

I did notice however, that there was no default target, as there were no emulators configured, the ones created, however, do not persist. Meaning one would create one per session, if need be. If you'd prefer I'd create another issue for this... issue.

**Comment by kamildzi** [comment]
I am glad the update worked for you. No need to apologize. :) 
I will close the issue now. 

Regarding your questions, please check following: 

### Case 1: 'sap' package issue
I think that this is a Tizen Studio bug. 
It seems to me, like it is happening only after installing 'Samsung Wearable Extension' package. 
Also I think that it is happening on specific platform versions only. 
It looks like Tizen Studio is trying to install an additional rpm package on the emulator to get the Extension working. 
If my understanding is correct, then there is little we can do about this problem in Tzen4Docker. 
You might be interested to check more details here: 
https://github.com/Samsung/Tizen.NET/issues/2
https://github.com/Samsung/Tizen.NET/issues/54

### Case 2: non-persistent emulators
Emulators should be preserved after the IDE closure. 
My first advice would be to look closely to the volume permissions: LOCAL_TIZEN_STUDIO_DATA_DIRECTORY, LOCAL_TIZEN_STUDIO_DIRECTORY, LOCAL_WORKSPACE. 
Tizen4Docker is using the volumes to store data in them and therefore - to make any IDE-related changes persistent. It might be a good idea to take a look inside these folders. If you have some leftovers from previous installations, you might also try to wipe everything inside. Then a totally clean installation should start. 
If above is not enough to solve this case, then please describe your problem and open a new issue.
